src/vmspawn/vmspawn-util.c

```diff
@@ -433,3 +434,36 @@ int vsock_fix_child_cid(int vhost_device_fd, unsigned *machine_cid, const char *
 
         return log_debug_errno(SYNTHETIC_ERRNO(EADDRNOTAVAIL), "Failed to assign a CID to the guest vsock");
 }
+
+char* escape_qemu_value(const char *s) {
+        const char *f;
+        char *r, *t;
+        size_t n;
+
+        assert(s);
+
+        /* QEMU requires that commas in arguments be escaped by doubling up the commas.
+         * See https://www.qemu.org/docs/master/system/qemu-manpage.html#options
+         * for more information.
+         *
+         * This function performs this escaping, returning an allocated string with the escaped value, or NULL if allocation failed. */
+
+        n = strlen(s);
+
+        r = new(char, n*2 + 1);
+        if (!r)
+                return NULL;
+
+        for (f = s, t = r; f < s + n; f++) {
+                if (*f == ',') {
+                        *t++ = ',';
+                        *t++ = ',';
+                } else
+                        *t++ = *f;
```

Comment:
one of the lines in the first branch and the 2nd branch actually come down to the same effect, since *f is ',' after all when the first branch is hit. Hence maybe just move the line from the 2nd branch out, drop the 2nd line from the first branch and thus get rid of one branch.

Reply:
a unit test for this helper would be cool btw.